src/NuGet.Core/NuGet.Protocol/Plugins/PluginDiscoverer.cs

```diff
@@ -140,6 +144,96 @@ private List<PluginFile> GetPluginFiles(CancellationToken cancellationToken)
             return files;
         }
 
+        /// <summary>
+        /// Gets auth plugins installed using dotnet tools. This is done by iterating through each file in directories found int eh
+        /// `PATH` environment variable.
+        /// The files are also expected to have a name that starts with `nuget-plugin-`
+        /// </summary>
+        /// <returns></returns>
+        private static List<string> GetNetToolsPluginFiles()
+        {
+            var pluginFiles = new List<string>();
+            var paths = Environment.GetEnvironmentVariable("PATH")?.Split(Path.PathSeparator) ?? Array.Empty<string>();
+
+            foreach (var path in paths)
+            {
+                if (Directory.Exists(path))
+                {
+                    var directoryInfo = new DirectoryInfo(path);
+                    var files = directoryInfo.GetFiles("nuget-plugin-*");
+
+                    foreach (var file in files)
+                    {
+                        if (IsValidPluginFile(file))
+                        {
+                            pluginFiles.Add(file.FullName);
+                        }
+                    }
+                }
+            }
+
+            return pluginFiles;
+        }
+
+        /// <summary>
+        /// Checks whether a file is a valid plugin file for windows/Unix.
+        /// Windows: It should be either .bat or  .exe
+        /// Unix: It should be executable
+        /// </summary>
+        /// <param name="fileInfo"></param>
+        /// <returns></returns>
+        private static bool IsValidPluginFile(FileInfo fileInfo)
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return fileInfo.Extension.Equals(".exe", StringComparison.OrdinalIgnoreCase) ||
+                       fileInfo.Extension.Equals(".bat", StringComparison.OrdinalIgnoreCase);
+            }
+            else
+            {
+                return fileInfo.Exists && (fileInfo.Attributes & FileAttributes.ReparsePoint) == 0 && IsExecutable(fileInfo);
+            }
+        }
+
+        /// <summary>
+        /// Checks whether a file is executable or not in Unix.
+        /// This is done by running bash code: `if [ -x {fileInfo.FullName} ]; then echo yes; else echo no; fi`
+        /// </summary>
+        /// <param name="fileInfo"></param>
+        /// <returns></returns>
+        private static bool IsExecutable(FileInfo fileInfo)
+        {
+#pragma warning disable CA1031 // Do not catch general exception types
+            try
+            {
+                string output;
+                using (var process = new Process())
+                {
+                    // Use a shell command to check if the file is executable
+                    process.StartInfo.FileName = "sh";
+                    process.StartInfo.Arguments = $"-c \"if [ -x {fileInfo.FullName} ]; then echo yes; else echo no; fi\"";
```

Comment:
Did you test this on mac?  Apparently OSX uses a very old version of bash (the default shell is zsh?), so some things that work on modern bash on linux don't work on mac, making cross platform shell scripting more difficult (since zsh often isn't isntalled on linux distros)

Reply:
I tested it on mac: both using zsh and sh, and it works.